extension/js/common/api/shared/api.ts

```diff
@@ -205,8 +206,23 @@ export class Api {
       // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
       timeout: typeof progress!.upload === 'function' || typeof progress!.download === 'function' ? undefined : 20000, // substituted with {} above
     };
-    const res = await Api.ajax(req, Catch.stackTrace());
-    return res as RT;
+    try {
+      const res = await Api.ajax(req, Catch.stackTrace());
+      return res as RT;
+    } catch (firstAttemptErr) {
+      const idToken = req.headers?.Authorization?.split(' ')[1];
+      if (ApiErr.isAuthErr(firstAttemptErr) && idToken) {
+        // force refresh token
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        const { email } = GoogleAuth.parseIdToken(idToken);
+        if (email) {
+          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+          req.headers!.Authorization = await GoogleAuth.googleApiAuthHeader(email, true);
+          return await Api.ajax(req, Catch.stackTrace()) as RT;
```

Comment:
As `Api` class uses `GoogleAuth` and `GoogleAuth` uses `Api` class, circular dependency error occurs.
So I'm thinking to move common method(parseIdToken, googleApiAuthHeader, googleAuthRefreshToken, googleAuthSaveTokens) to new common class(such as GoogleAuthExtension)

@sosnovsky Do you have better suggestion?

Reply:
I think proposed solution should be good, we can call this new class `GoogleAuthHelper`

Reply:
Thanks